src/dmd/backend/inliner.d

```diff
@@ -0,0 +1,565 @@
+/**
+ * Implementation of the
```

Comment:
Reduced one of the linker errors in the testsuite:
```d
// config.d
import parseoptions;

struct Config
{
    bool initialize()
    {
        return initConfigOptions(this, "gcopt");
    }

    string errorName() { return "GC"; }
}
```
```d
// parseoptions.d
bool initConfigOptions(CFG)(CFG cfg, string )
{
    string parse(string opt)
    {
        if (parseOptions(cfg, opt))
            return "err";
        return null; 
    }
    string s;
    return s is null;
}


bool parseOptions(CFG)(CFG cfg, string )
{
    string errName = cfg.errorName;
    optError(errName);
    return true;
}

bool optError(const(char)[] )
{
}
```

Run the following:
```
./generated/linux/release/64/dmd -lib -O -inline config.d
nm config.a | grep "U _TMP"
```
What looks to be happening, is a function is being inlined that references a local (static) string symbol.  The destination object can't resolve the symbol.

Reply:
Yes, you are right! Fixing...